actionpack/test/dispatch/ssl_test.rb

```diff
@@ -58,6 +70,10 @@ def assert_redirected(redirect: {}, deprecated_host: nil, deprecated_port: nil,
     assert_redirected
   end
 
+  test 'http POST is redirected to https with status 307' do
+    assert_post_redirected
```

Comment:
@kaspth if we explicitly pass the redirect status here, will it not render the test useless as the app will always assign this status code. We want to test that when no status is explicitly specified, it responds with 307 status code for non-GET or non-HEAD requests. Or am I missing something here?